Add rocsolver logging tests

cgmb opened the pull request:
I've split the tests from #370 into this PR for administrative reasons. Marked as draft, as I've yet to address the comments from that PR.

See SWDEV-309338.

tfalders commented on clients/gtest/logging_gtest.cpp:
```diff
@@ -0,0 +1,415 @@
+    {
+        if(fs::exists(log_filepath))
+        {
+            if(HasFailure() && std::getenv("ROCSOLVER_TEST_DEBUG"))
```
Why &&? Is this to avoid file clutter even when ROCSOLVER_TEST_DEBUG is enabled?

tfalders reviewed (approved):
LGTM!

cgmb commented:
I'm satisfied with this PR as-is, so I've enabled auto-merge. It will automatically merge itself once there's two approvals and all `Required` tests have passed.

jzuniga-amd reviewed (approved):
LGTM. Good job Cory.

cgmb commented:
Windows is failing on the PSDB builds. Mea culpa.

The Windows CI was down but I went ahead and merged this anyway. In addressing the review comments, I added a few tests to check we are returning proper error codes when the rocsolver API is misused. I named those tests things like `checkin_misc_LOGGING.begin_twice_failure`. Unfortunately, if `error` or `fail` appear anywhere in the rocsolver-test output, then rtest.py will mark the test suite as a failure.

In my opinion, this is a misfeature in rtest.py. There's no need to parse the log output when rocsolver-test has two explicit mechanisms for signaling test failures (the process exit code and the xml test results file). The rtest.py behaviour also caused a similar problem in rocBLAS a few weeks ago.

I'll fix the copy of rtest.py in the rocSOLVER and hipSOLVER repos, and open a PR to update the canonical version.
